pkg/chaosdaemon/tasks/chaosonprocess.go

```diff
@@ -0,0 +1,252 @@
+// the Recover function will return a ErrNotImplement("ChaosCanRecover") error.
+type ChaosOnProcessManager struct {
+	taskManager TaskManager
+	processMap  map[PID]ChaosOnProcess
+
+	logger logr.Logger
+}
+
+func NewChaosOnProcessManager(logger logr.Logger) ChaosOnProcessManager {
+	return ChaosOnProcessManager{
+		NewTaskManager(),
+		make(map[PID]ChaosOnProcess),
+		logger,
+	}
+}
+
+func (cm ChaosOnProcessManager) CopyTaskManager() TaskManager {
+	tm := NewTaskManager()
+	for uid, task := range cm.taskManager.TaskMap {
+		tm.TaskMap[uid] = task
+	}
+	return tm
+}
+
+func (cm ChaosOnProcessManager) CopyProcessMap() map[PID]ChaosOnProcess {
+	pm := make(map[PID]ChaosOnProcess)
+	for pid, chaosOnProcess := range cm.processMap {
+		cm.processMap[pid] = chaosOnProcess
+	}
+	return pm
+}
+
+func (cm ChaosOnProcessManager) GetWithUID(id UID) (interface{}, error) {
+	return cm.taskManager.GetWithUID(id)
+}
+
+func (cm ChaosOnProcessManager) GetWithPID(pid PID) (ChaosOnProcess, error) {
+	p, ok := cm.processMap[pid]
+	if !ok {
+		return nil, ChaosErr.NotFound("PID")
+	}
+	return p, nil
+}
+
+func (cm ChaosOnProcessManager) GetUIDsWithPID(pid PID) []UID {
+	return cm.taskManager.GetWithPID(pid)
+}
+
+// Update the task config with a same UID and PID .
+// If it comes a UID not injected , Update will return ChaosErr.NotFound("UID").
+// If it comes the import PID of task do not equal to the last one,
+// Update will return ErrUpdateTaskWithPIDChanges.
+func (cm ChaosOnProcessManager) Update(uid UID, pid PID, config TaskToProcess) error {
+	oldTask, err := cm.taskManager.UpdateTask(uid, NewTask(pid, config))
+	if err != nil {
+		return err
+	}
+	err = cm.commit(uid, pid)
+	if err != nil {
+		_, _ = cm.taskManager.UpdateTask(uid, oldTask)
+		return err
+	}
+	return nil
+}
+
+// Create the first task on a process,
+// the New function of TaskToProcess:NewChaosOnProcess will only be used here.
+// immutableValues is only the import parameter of New function in TaskToProcess:NewChaosOnProcess.
+// If it comes a PID are already be injected ,
+// Create will return ChaosErr.ErrDuplicateEntity.
+func (cm ChaosOnProcessManager) Create(uid UID, pid PID, config TaskToProcess, immutableValues interface{}) error {
+	if _, ok := cm.processMap[pid]; ok {
+		return errors.Wrapf(ChaosErr.ErrDuplicateEntity, "create")
+	}
+
+	err := cm.taskManager.AddTask(uid, NewTask(pid, config))
+	if err != nil {
+		return err
+	}
+
+	processTask, err := config.New(immutableValues)
+	if err != nil {
+		_ = cm.taskManager.RecoverTask(uid)
+		return errors.Wrapf(err, "fork time skew : %v", config)
+	}
+
+	cm.processMap[pid] = processTask
+	err = cm.commit(uid, pid)
+	if err != nil {
+		_ = cm.taskManager.RecoverTask(uid)
+		delete(cm.processMap, pid)
+		return errors.Wrapf(err, "update new process")
+	}
+	return nil
+}
+
+// Apply the task when the target pid of task is already be Created.
+// If it comes a UID injected , Apply will return ChaosErr.ErrDuplicateEntity.
```

Comment:
seems that it does not return `ErrDuplicateEntity` error

Reply:
It will be returned by `AddTask`.